pkg/agent/datapath/policyBridge.go

```diff
@@ -435,17 +435,21 @@ func (p *PolicyBridge) AddMicroSegmentRule(rule *EveroutePolicyRule, direction u
 
 	// Install the rule in policy table
 	ruleFlow, err := policyTable.NewFlow(ofctrl.FlowMatch{
-		Priority:   uint16(rule.Priority),
-		Ethertype:  PROTOCOL_IP,
-		IpDa:       ipDa,
-		IpDaMask:   ipDaMask,
-		IpSa:       ipSa,
-		IpSaMask:   ipSaMask,
-		IpProto:    rule.IPProtocol,
-		TcpSrcPort: rule.SrcPort,
-		TcpDstPort: rule.DstPort,
-		UdpSrcPort: rule.SrcPort,
-		UdpDstPort: rule.DstPort,
+		Priority:       uint16(rule.Priority),
+		Ethertype:      PROTOCOL_IP,
+		IpDa:           ipDa,
+		IpDaMask:       ipDaMask,
+		IpSa:           ipSa,
+		IpSaMask:       ipSaMask,
+		IpProto:        rule.IPProtocol,
+		TcpSrcPort:     rule.SrcPort,
+		TcpSrcPortMask: rule.SrcPortMask,
+		TcpDstPort:     rule.DstPort,
+		TcpDstPortMask: rule.DstPortMask,
+		UdpSrcPort:     rule.SrcPort,
+		UdpSrcPortMask: rule.SrcPortMask,
+		UdpDstPort:     rule.DstPort,
+		UdpDstPortMask: rule.DstPortMask,
```

Comment:
If the protocol is TCP, the port is for TCP, but the UdpSrcPort/DstPort/Mask will be set as the same as TCP, will this a problem?  And vise versa.

Reply:
It's no problem, this is a validated feature in the previous version of the datapath.

Reply:
if the protocol is tcp then udp port and mask are not necessary to set for this flow, right?

Reply:
Same as the other field, for the field which is not defined in policyrule, we wouldn't add this field to match fields.  Controller define which filed is need, and sets this field.